src/libraries/System.Net.Http/src/System/Net/Http/StringContent.cs

```diff
@@ -45,8 +46,31 @@ public StringContent(string content, Encoding? encoding)
         /// <param name="encoding">The encoding to use for the content.</param>
         /// <param name="mediaType">The media type to use for the content.</param>
         public StringContent(string content, Encoding? encoding, string mediaType)
-            : this(content, encoding, new MediaTypeHeaderValue(mediaType ?? DefaultMediaType, (encoding ?? DefaultStringEncoding).WebName))
+            : base(GetContentByteArray(content, encoding))
         {
+            Debug.Assert(DefaultStringEncoding.WebName == "utf-8");
+
+            encoding ??= DefaultStringEncoding;
+            mediaType ??= DefaultMediaType;
+
+            // Avoid allocating MediaTypeHeaderValue and related objects for common media types.
+            if (ReferenceEquals(encoding, DefaultStringEncoding))
+            {
+                string? knownValue = mediaType switch
+                {
+                    "text/plain" => "text/plain; charset=utf-8",
+                    "application/json" => "application/json; charset=utf-8",
+                    _ => null
+                };
+
+                if (knownValue is not null)
+                {
+                    Headers.TryAddWithoutValidation(KnownHeaders.ContentType.Descriptor, knownValue);
```

Comment:
Will `Header.ContentType` be populated in this case?

Reply:
Yes, behavior-wise as far as the user can see, the two should be the same.

The `ContentType` property uses the underlying dictionary as the source of truth / storage, and will lazily parse the value when first accessed.